components/permalink_view/permalink_view.jsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import {Link} from 'react-router-dom';
+
+import ChannelHeader from 'components/channel_header';
+import PostView from 'components/post_view';
+import {emitPostFocusEvent} from 'actions/global_actions.jsx';
+
+export default class PermalinkView extends React.PureComponent {
+    static propTypes = {
+        channelId: PropTypes.string,
+        channelName: PropTypes.string,
+        returnTo: PropTypes.string.isRequired,
+        teamName: PropTypes.string,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {valid: false};
+    }
+
+    componentDidMount() {
+        this.doPermalinkEvent(this.props);
+        document.body.classList.add('app__body');
+    }
+
+    componentWillUnmount() {
+        document.body.classList.remove('app__body');
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.doPermalinkEvent(nextProps);
+    }
+
+    doPermalinkEvent = async (props) => {
+        this.setState({valid: false});
+        const postId = props.match.params.postid;
+        await emitPostFocusEvent(postId, this.props.returnTo);
+        this.setState({valid: true});
+    }
+
+    isStateValid = () => {
+        return this.state.valid && this.props.channelId && this.props.teamName;
+    }
+
+    render() {
+        const {
+            channelId,
+            channelName,
+            match,
```

Comment:
I really don't understand how this is not throwing an ESLint error, what I meant is that you should define `match` in the above
```js
static propTypes = {
...
}
```

Reply:
Ah I see, and also we have similar case in several components as well. I'll submit separate PR to define `match` to other components.

Reply:
`match` is ignored in our `.eslintrc.json`,
```
"react/prop-types": [2, { "ignore": ["match", "location", "history", "component"] }],
```

Reply:
hmm, interesting.. but yes I do prefer to have it explicitly